crates/proto/src/xfer/dns_exchange.rs

```diff
@@ -161,52 +238,78 @@ where
     }
 }
 
-/// A wrapper for a future DnsExchange connection
-pub struct DnsExchangeConnect<F, S, R>(DnsExchangeConnectInner<F, S, R>)
+/// A wrapper for a future DnsExchange connection.
+///
+/// DnsExchangeConnect is clonable, making it possible to share this if the connection
+///  will be shared across threads.
+///
+/// The future will return a tuple of the DnsExchange (for sending messages) and a background
+///  for running the background tasks. The background is optional as only one thread should run
+///  the background. If returned, it must be spawned before any dns requests will function.
+pub struct DnsExchangeConnect<F, S, R>(Arc<Mutex<DnsExchangeConnectInner<F, S, R>>>)
```

Comment:
@hawkw I'm reviewing the mutexes around the system. This one is needed to allow the connect to be clone between different tasks. This is necessary for NameServer and NameServerPool. It's only used during initial connection, not to drive the background I/O.

Reply:
FYI, I was able to drop this mutex, @hawkw... :)